common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -164,10 +182,7 @@ private AppShufflePartitionInfo getOrCreateAppShufflePartitionInfo(
           // current incoming one
           int latestShuffleMergeId = appShuffleMergePartitionsInfo.shuffleMergeId;
           if (latestShuffleMergeId > shuffleMergeId) {
-            throw new StaleBlockPushException(String.format("Rejecting shuffle blocks push request"
-              + " for shuffle %s with shuffleMergeId %s for application %s_%s as a higher"
-              + " shuffleMergeId %s request is already seen", shuffleId, shuffleMergeId,
-              appShuffleInfo.appId, appShuffleInfo.attemptId, latestShuffleMergeId));
```

Comment:
log the error message?

Reply:
Could we also log for the case of:
```
if (dataFile.exists()) {
  return null;
}
```

Reply:
Added exception message into `BlockPushNonFatalFailure`, and let the existing logic in ErrorHandler to handle which level to log these.